### DISCUSSION TITLE
Como integrar o Azure AD autenticando no GitHub

### DISCUSSION THREAD

**Discussion by fabiojrmello** [community/community#69649] [Actions]
### Select Topic Area

General

### Body

Prezados, estou em busca da documentação, á onde eu possa inserir o Azure AD do Office 365 autenticando no GitHub Actions, já realizei buscar do item na pagina da Microsoft e do GitHub, só encontro a documentação a onde ensina inserir a mesma usando a organização com o GitHub Enterprise, só que atualmente a minha organização é GitHub Action, alguém consegue me auxiliar com a mesma.

**Comment by santoshvandari** [comment]
Hello

You're looking to integrate Azure AD (Active Directory) authentication with GitHub Actions. While there might not be direct documentation for integrating Azure AD with GitHub Actions specifically, you can achieve this by setting up Azure AD authentication for your GitHub repositories. This would allow users to authenticate using their Azure AD credentials when accessing your GitHub Actions workflows.

Here's a general approach to integrate Azure AD authentication with GitHub Actions:

Configure Azure AD: Set up Azure AD to work with your GitHub organization or repository. You'll need to configure Azure AD to provide authentication services. Microsoft's documentation can guide you through the process of setting up Azure AD for your organization.

GitHub Repository Permissions: Ensure that the users you want to grant access to your GitHub Actions workflows are added as collaborators or have the necessary permissions on your GitHub repository. This can be done through the repository settings on GitHub.

GitHub Actions Workflow: Create or modify your GitHub Actions workflow to require authentication. You can use GitHub Secrets to securely store sensitive information, such as Azure AD credentials or tokens, and use them within your workflows.

Access Control: Within your GitHub Actions workflow, you can implement access control logic that checks if a user is authenticated via Azure AD before allowing them to perform specific actions.

Please note that this is a high-level overview, and the specifics may vary based on your organization's needs and the structure of your GitHub Actions workflows.

Additionally, consider reaching out to Microsoft Azure support or GitHub Support for more detailed assistance in setting up Azure AD authentication for your GitHub Actions environment, as they can provide guidance tailored to your specific setup.

I hope this helps point you in the right direction for integrating Azure AD authentication with GitHub Actions.
